colorcoder/ColorPairImpl.java

```diff
@@ -0,0 +1,29 @@
+package colorcoder;
+
+public class ColorPairImpl {
+
+	public ColorPair getColorFromPairNumber(int pairNumber) {
+		int zeroBasedPairNumber = pairNumber - 1;// 3
+		MajorColor majorColor = MajorColor.fromIndex(zeroBasedPairNumber / MajorColor.values().length);

```

Comment:
zeroBasedPairNumber / MajorColor.values().length needs to be corrected to use MinorColor values length. Since the length of major and minor colors are same so the bug does not manifest. But when the length is different then this code will pose problem. You may think of test cases which can cover this code and verify its correctness.